client/src/main/java/org/eclipse/hono/client/impl/AmqpAdapterClientFactoryImpl.java

```diff
@@ -80,7 +83,7 @@ protected void onDisconnect() {
     @Override
     public Future<TelemetrySender> getOrCreateTelemetrySender() {
 
-        final String cacheKey = TelemetrySenderImpl.getTargetAddress(tenantId, null);
+        final String cacheKey = AddressHelper.getTargetAddress(TelemetryConstants.TELEMETRY_ENDPOINT, tenantId, null, connection.getConfig().getAddressRewriteRule());
```

Comment:
the `AmqpAdapterClientFactory` is supposed to be used by devices and gateways to connect to the AMQP adapter. IMHO we do not need to rewrite addresses in that case, do we?